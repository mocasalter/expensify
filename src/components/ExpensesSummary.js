import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import numeral from 'numeral'
import selectExpenses from '../selectors/expenses.js'
import selectExpensesTotal from '../selectors/expenses-total'

// Destructure 3 items from the props object. This gives us easy access the props from below in mapStateToProps
export const ExpensesSummary = ({ expenseCount, expensesTotal, expenseCountHidden }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses' 
  const expenseWordsHidden = expenseCountHidden === 1 ? 'expense is' : 'expenses are' 
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

  return (
    <div className="page-header">
      <div className="content-container">
        <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
        {expenseCountHidden > 0 && (
          <p class="page-header__subtext">{expenseCountHidden} {expenseWordsHidden} hidden by the applied filters</p>
        )}
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)

    return {
        expenseCount: visibleExpenses.length,
        expenseCountHidden: (state.expenses.length - visibleExpenses.length),
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)