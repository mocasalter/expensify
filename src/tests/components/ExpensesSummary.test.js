import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary.js'

test('should correctly render ExpensesSummarry with 1 expense ', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
    expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpensesSummarry with multiple expense ', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={235998} />)
    expect(wrapper).toMatchSnapshot()
})
