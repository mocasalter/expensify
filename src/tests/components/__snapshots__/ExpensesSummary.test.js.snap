// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummarry with 1 expense  1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </p>
  <div>
    <Link
      replace={false}
      to="/create"
    >
      Add Expense
    </Link>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummarry with multiple expense  1`] = `
<div>
  <p>
    Viewing 
    23
     
    expenses
     totalling 
    $2,359.98
  </p>
  <div>
    <Link
      replace={false}
      to="/create"
    >
      Add Expense
    </Link>
  </div>
</div>
`;
